scripts/prepare_sector_network.py

```diff
@@ -1121,14 +1126,61 @@ def add_generation(n, costs):
             * costs.at[generator, "VOM"],  # NB: VOM is per MWel
             capital_cost=costs.at[generator, "efficiency"]
             * costs.at[generator, "fixed"],  # NB: fixed cost is per MWel
-            p_nom_extendable=True,
+            p_nom_extendable=(
+                True
+                if generator
+                in snakemake.params.electricity.get("extendable_carriers", dict()).get(
+                    "Generator", list()
+                )
+                else False
+            ),
+            p_nom=(
+                existing_capacities[generator] / existing_efficiencies[generator]
+                if not existing_capacities == 0
+                else 0
+            ),  # NB: existing capacities are MWel
+            p_max_pu=(
+                0.7 if carrier == "uranium" else 1
+            ),  # be conservative for nuclear (maintenance or unplanned shut downs)
```

Comment:
For consistency, it would be good to use this value in `add_existing_baseyear`as well.

https://github.com/PyPSA/pypsa-eur/blob/728eacf5589ed2ede90dfe59fd371c66739aa835/scripts/add_existing_baseyear.py#L319-L337

It would be even better to use the country specific values from [data/nuclear_p_max_pu.csv](https://github.com/PyPSA/pypsa-eur/blob/728eacf5589ed2ede90dfe59fd371c66739aa835/data/nuclear_p_max_pu.csv).

@daniel-rdt